iam/oauth.go

```diff
@@ -163,6 +163,19 @@ func getToken(grantType OAuthGrantType, endpoint string) (token adal.OAuthTokenP
 	return
 }
 
+// GetResourceManagementTokenHybrid retrieves auth token for hybrid environment
+func GetResourceManagementTokenHybrid(activeDirectoryEndpoint, tokenAudience, tenantID, clientID, clientSecret string) (adal.OAuthTokenProvider, error) {
```

Comment:
Is it possible to login through device auth?

Reply:
If you mean using getDeviceToken method, I am afraid we cannot use it. This is because getDeviceToken method uses oauthConfig parameters that is defined in the init() method. oauthConfig parameter is created by using hard coded value of azure.PublicCloud.ActiveDirectoryEndpoint. We should use a customized value for ActiveDirectoryEndpoint in Stack environment.
However, I think it is possible to use adal.InitiateDeviceAuth method to generate token (I have not tried it, though). But I am not sure if there is any benefit in using InitiateDeviceAuth method instead of NewServicePrincipalToken in hybrid environments.

Reply:
Okok, could you take a closer look at this?
Or if there is no point in device auth for hybrid, please clarify in the readme this option is not enabled in the hybrid samples.

---
In reply to: [177173861](https://github.com/Azure-Samples/azure-sdk-for-go-samples/pull/144#discussion_r177173861) [](ancestors = 177173861)